[문승준] Mission 4. 로그인 기능 구현

palza4dev opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
- 로그인 기능 view 작성
		 
## 수정 사항들 자세한 내용
- email이나 password 키가 없으면 키에러 메시지 반환
- input_email이 올바르지 않으면 invalid_email 메시지 반환
- input_email의 user password와 input_password가 같으면 로그인 성공
- `/login` 으로 url 연결 완료

## 기타 질문 및 특이 사항
- 없음

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
조건을 만족한다면 각 빈칸에 x 표시 해주세요
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

## POSTMAN을 활용한 로그인 테스트
![스크린샷 2021-09-28 오후 5 55 30](https://user-images.githubusercontent.com/72376931/135056315-1f712af9-ca51-43b6-9d85-b5201e42e572.png)

palza4dev committed:
<Add> 로그인 구현을 위한 LogInView 작성
- 잘못된 email에는 Invalid_email 에러 처리
- input_email user의 비밀번호와 input_password가 일치하면 로그인 성공
- 입력키가 없으면 key error 메시지 처리
- urls.py path에 /login 추가

lordmyshepherd commented on students/seungjun/users/views.py:
```diff
@@ -41,3 +41,23 @@ def post(self, request):
+
+class LogInView(View):
+    def post(self,request):
+        data = json.loads(request.body)
+        try:
+            input_email    = data['email']
+            input_password = data['password']
```
변수명은 간결하고 명확하게 선언해주세요. `input_` 과 같은 용어는 불필요합니다~
또한 json.loads() 함수를 호출하는 경우에 decode error가 발생할 수 있으므로 `try`, `except`문 안에 추가해주세요!

```suggestion

        try:
            data      = json.loads(request.body)
            email    = data['email']
            password = data['password']
```

palza4dev replied:
data 변수는 try문 안에서 선언하고, email과 password 변수명 수정하였습니다.

palza4dev committed:
<Modify> LogInView class 일부 변수 수정
- json.loads 함수의 decode error에 대비하여 try문 안으로 선언 위치 이동
- 로그인을 위해 입력된 email과 password의 변수명을 간결하게

kyeongraekim commented on students/seungjun/users/views.py:
```diff
@@ -41,3 +41,23 @@ def post(self, request):
+
+            user = User.objects.get(email=email)
+            
+            if user.password == password:
```
기능상에서는 문제 없어 보이는 코드이나 가독성 면에서 제한 조건들을 다 통과하고 나면 제일 마지막에 로그인이 성공하게 되는 로직이 가독성과 로직 플로우상 흐름이  좋아 보입니다. 한 번 수정을 고려해보세요!

palza4dev replied:
패스워드가 일치하지 않으면 에러를 반환하도록 수정하고, 로그인 성공 로직은 마지막으로 옮겼습니다

kyeongraekim commented on students/seungjun/users/views.py:
```diff
@@ -7,8 +7,8 @@
     def post(self, request):
-        data = json.loads(request.body)
         try:
+            data = json.loads(request.body)
```
새로 추가 해주신 코드 내용의 변수명 정렬이 조금 아쉽습니다. 간단히 수정해주세요!

palza4dev replied:
다른 변수들과 맞추어 재정렬 했습니다!

kyeongraekim reviewed (changes requested):
안녕하세요 승준님! httpie가 아닌 포스트맨으로 직접 통신 시험도 잘 해주셨습니다.
나머지 추가 수정 사항 아래 내용 확인해주시고 수정해주세요~

palza4dev committed:
<Modify> LogInView 일부 로직 수정 및 SignUpView 코드 정렬
- LogInView password 검증은 미일치시 에러를 반환하도록 수정
- 모든 제한 조건을 통과하고 성공하도록 로그인 흐름을 변경
- SignUpView data 변수 align 수정 완료

kyeongraekim commented:
LGTM!~ 다음 단계 이어서 진행해주세요~
